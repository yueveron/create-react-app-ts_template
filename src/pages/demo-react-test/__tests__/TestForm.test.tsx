import { act } from 'react-dom/test-utils';
import { render, screen, cleanup, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';
import MyForm from '../pages/TestFormik';

afterEach(cleanup);

test('test Formik form', async () => {
  act(() => {
    render(<MyForm />);
  });

  const user = userEvent.setup();
  // use type() to insert some text into an input field or textarea
  await user.type(screen.getByRole('textbox', { name: /first name/i }), 'John');
  await user.type(screen.getByRole('textbox', { name: /last name/i }), 'Dee');
  await user.type(
    screen.getByRole('textbox', { name: /email/i }),
    'john.dee@someemail.com'
  );
  await user.click(screen.getByRole('button', { name: /submit/i }));

  const titleEle = screen.getByTestId('counter');
  expect(titleEle).toHaveTextContent('Init');
  await waitFor(() => {
    expect(titleEle).toHaveTextContent('John');
  });
});
