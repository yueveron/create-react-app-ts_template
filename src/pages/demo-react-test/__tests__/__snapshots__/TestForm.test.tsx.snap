// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Formik Test Cases test Formik form: test-formik 1`] = `
<div>
  <div>
    <h1>
      Sign Up
    </h1>
    <span
      data-testid="counter"
    >
      John
    </span>
    <form
      action="#"
      data-testid="test-form"
    >
      <label
        for="firstName"
      >
        First Name
      </label>
      <input
        data-testid="firstName"
        id="firstName"
        name="firstName"
        placeholder="Jane"
        value="John"
      />
      <label
        for="lastName"
      >
        Last Name
      </label>
      <input
        id="lastName"
        name="lastName"
        placeholder="Doe"
        value="Dee"
      />
      <label
        for="email"
      >
        Email
      </label>
      <input
        id="email"
        name="email"
        placeholder="jane@acme.com"
        type="email"
        value="john.dee@someemail.com"
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
